simulator.go

```diff
@@ -250,6 +251,22 @@ func (h *simulatorAPIHandler) CheckTimeout() {
 }
 
 func (h *simulatorAPIHandler) terminateContainer(id string, w http.ResponseWriter) {
+
+	h.lock.Lock()
```

Comment:
I don't think this was actually racy... or if it was, I don't think you fixed it the right way. Even before this change, the callers of `terminateContainer` already held this lock, see e.g. L581.